src/main/java/org/opensearch/jobscheduler/model/ExtensionJobParameter.java

```diff
@@ -0,0 +1,141 @@
+
+/**
+ * A {@link Writeable} ScheduledJobParameter used to transport job parameters between OpenSearch and Extensions
+ *
+ */
+public class ExtensionJobParameter implements ScheduledJobParameter, Writeable {
+
+    enum ScheduleType {
+        CRON,
+        INTERVAL
+    }
+
+    private String extensionJobName;
+
+    private Schedule extensionJobSchedule;
+
+    private Instant lastUpdateTime;
+
+    private Instant enabledTime;
+
+    private boolean isEnabled;
+
+    private Long lockDurationSeconds;
+
+    private Double jitter;
+
+    public ExtensionJobParameter(
+        String extensionJobName,
+        Schedule extensionJobSchedule,
+        Instant lastUpdateTime,
+        Instant enabledTime,
+        boolean isEnabled,
+        Long lockDurationSeconds,
+        Double jitter
+    ) {
+        this.extensionJobName = extensionJobName;
+        this.extensionJobSchedule = extensionJobSchedule;
+        this.lastUpdateTime = lastUpdateTime;
+        this.enabledTime = enabledTime;
+        this.isEnabled = isEnabled;
+        this.lockDurationSeconds = lockDurationSeconds;
+        this.jitter = jitter;
+    }
+
+    public ExtensionJobParameter(ScheduledJobParameter jobParameter) {
+        // Convert job Parameter into writeable ExtensionJobParameter
+        this(
+            jobParameter.getName(),
+            jobParameter.getSchedule(),
+            jobParameter.getLastUpdateTime(),
+            jobParameter.getEnabledTime(),
+            jobParameter.isEnabled(),
+            jobParameter.getLockDurationSeconds(),
+            jobParameter.getJitter()
+        );
+    }
+
+    public ExtensionJobParameter(StreamInput in) throws IOException {
+        this.extensionJobName = in.readString();
+        if (in.readEnum(ExtensionJobParameter.ScheduleType.class) == ScheduleType.CRON) {
+            this.extensionJobSchedule = new CronSchedule(in);
+        } else {
+            this.extensionJobSchedule = new IntervalSchedule(in);
+        }
+        this.lastUpdateTime = in.readInstant();
+        this.enabledTime = in.readInstant();
+        this.isEnabled = in.readBoolean();
+        this.lockDurationSeconds = in.readLong();
+        this.jitter = in.readDouble();
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeString(this.extensionJobName);
+        if (this.extensionJobSchedule instanceof CronSchedule) {
+            out.writeEnum(ScheduleType.CRON);
+        } else {
+            out.writeEnum(ScheduleType.INTERVAL);
+        }
+        this.extensionJobSchedule.writeTo(out);
+        out.writeInstant(this.lastUpdateTime);
+        out.writeInstant(this.enabledTime);
+        out.writeBoolean(this.isEnabled);
+        out.writeLong(this.lockDurationSeconds);
+        out.writeDouble(this.jitter);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        // no op
+        return null;
```

Comment:
This seems dangerous.  Is there a better "empty builder" constant we can return here?